modules/reitit-core/src/reitit/impl.cljc

```diff
@@ -155,3 +157,53 @@
 
 (defn strip-nils [m]
   (->> m (remove (comp nil? second)) (into {})))
+
+;;
+;; Path-parameters, see https://github.com/metosin/reitit/issues/75
+;;
+
+(defn url-encode [s]
+  (some-> s
+          #?(:clj  (URLEncoder/encode "UTF-8")
+             :cljs (js/encodeURIComponent))
+          (.replace "+" "%20")))
+
+(defn url-decode [s]
+  (some-> s #?(:clj  (URLDecoder/decode "UTF-8")
+               :cljs (js/decodeURIComponent))))
+
+(defprotocol IntoString
+  (into-string [_]))
+
+(extend-protocol IntoString
+  #?(:clj  String
+     :cljs string)
+  (into-string [this] this)
+
+  #?(:clj  clojure.lang.Keyword
+     :cljs cljs.core.Keyword)
+  (into-string [this]
+    (let [ns (namespace this)]
+      (str ns (if ns "/") (name this))))
+
+  #?(:clj  Boolean
+     :cljs boolean)
+  (into-string [this] (str this))
+
+  #?(:clj  Number
+     :cljs number)
+  (into-string [this] (str this))
+
+  #?(:clj  Object
+     :cljs object)
+  (into-string [this] (str this)))
+
+(defn path-params
+  "shallow transform of the path-param values into strings"
+  [params]
+  (persistent!
```

Comment:
Did you benchmark using transients here? Because I recall that transients for small maps (TransientArrayMap/PersistentArrayMap) are slower than persistents, because Rich Hickey has optimized the latter but not applied the same optimizations to the former.

(Which should be fixed, but I've never gotten around to opening an issue or submitting a patch.)

Reply:
Oh, true that. non-transient is faster with small sizes, which here is the case